stan/math/prim/scal/fun/log_modified_bessel_first_kind.hpp

```diff
@@ -0,0 +1,104 @@
+#ifndef STAN_MATH_PRIM_SCAL_FUN_LOG_MODIFIED_BESSEL_FIRST_KIND_HPP
+#define STAN_MATH_PRIM_SCAL_FUN_LOG_MODIFIED_BESSEL_FIRST_KIND_HPP
+
+#include <stan/math/prim/scal/err/check_not_nan.hpp>
+#include <stan/math/prim/scal/err/check_nonnegative.hpp>
+#include <stan/math/prim/scal/err/check_greater_or_equal.hpp>
+#include <stan/math/prim/scal/fun/is_inf.hpp>
+#include <stan/math/prim/scal/fun/is_nan.hpp>
+#include <stan/math/prim/scal/fun/lgamma.hpp>
+#include <stan/math/prim/scal/fun/log_sum_exp.hpp>
+#include <stan/math/prim/scal/fun/log1p.hpp>
+#include <stan/math/prim/scal/fun/square.hpp>
+#include <limits>
+
+namespace stan {
+  namespace math {
+
+    /* Log of the modified Bessel function of the first kind,
+     * which is better known as the Bessel I function. See
+     * modified_bessel_first_kind.hpp for the function definition.
+     *
+     * @param v Order, can be a non-integer but must be at least -1
+     * @param z Real non-negative number
+     * @throws if either v or z is NaN
+     * @throws if z is negative
+     * @throws if v is less than -1
+     * @return log of Bessel I function
+     */
+    template <typename T1, typename T2>
+    inline typename boost::math::tools::promote_args<T1, T2, double>::type
+    log_modified_bessel_first_kind(const T1 v, const T2 z) {
+      check_not_nan("log_modified_bessel_first_kind", "v", v);
+      check_not_nan("log_modified_bessel_first_kind", "z", z);
+      check_nonnegative("log_modified_bessel_first_kind", "z", z);
+      check_greater_or_equal("log_modified_bessel_first_kind", "v", v, -1);
+
+      using std::log;
+      using stan::math::log_sum_exp;
```

Comment:
Symbols in the same namespace get brought in automatically---so you don't need the `stan::math::` using statements.  They won't hurt anything being there.